cppcache/include/geode/DataInput.hpp

```diff
@@ -313,7 +313,53 @@ class APACHE_GEODE_EXPORT DataInput {
         // empty string
         break;
       // TODO: What's the right response here?
-      default:
+      case internal::DSCode::FixedIDDefault:
```

Comment:
Wow I really really don't like this.  Is there anything we can do to fix the warning _besides_ adding ~65 case statements?  This makes the code less readable, not more IMO.

Reply:
I actually like it since it forces you to acknowledge all possible switches for an enum (which is enumerable).  That way you don't accidentally go to the default when adding a new enum.  If the general consensus is we prefer not to change `-Wno-switch-enum` then we can revert that commit.  I could make it worse by adding `break;` statements in between all the cases :-p

Reply:
Sorry, I don't think the chance of getting a warning for free is worth the cost of some egregious cluttering of the code.  We're only using 4 of the 61 possible values for a DSCode, so it was kind of an abuse of a switch statement to begin with.  This should probably just be a 4-way if-else block.

Reply:
I'll switch it to an if-else block and we'll see if that cleans up better

Reply:
Just gonna leave this here, cause I spent the time & effort to figure it out.  Probably slightly better performance than the if-else approach, but I'm still not sure about readability:

```
  template <class CharT = char, class... Tail>
  inline std::basic_string<CharT, Tail...> readString() {
    std::basic_string<CharT, Tail...> value;
    std::map<internal::DSCode, std::function<void(std::string&)> > readers;

    readers.insert(std::make_pair(
        internal::DSCode::CacheableString,
        [=](std::string& val) { this->readJavaModifiedUtf8(val); }));

    readers.insert(
        std::make_pair(internal::DSCode::CacheableStringHuge,
                       [=](std::string& val) { this->readUtf16Huge(val); }));

    readers.insert(
        std::make_pair(internal::DSCode::CacheableASCIIString,
                       [=](std::string& val) { this->readAscii(val); }));

    readers.insert(
        std::make_pair(internal::DSCode::CacheableASCIIStringHuge,
                       [=](std::string& val) { this->readAsciiHuge(val); }));

    auto type = static_cast<internal::DSCode>(read());
    auto it = readers.find(static_cast<internal::DSCode>(read()));

    if (it != readers.end()) {
      it->second(value);
    }
    return value;
  }

```

Reply:
I checked the `Allow edits and access to secrets by maintainers` so feel free to add this as commit and see if it works (I'm assuming the commit will go to the branch and trigger the [CI pipeline](https://github.com/moleske/geode-native/actions) I've been playing around with)

Reply:
Per some other avenue feedback from @pdxcodemonkey and @mreddington , I'll take some time to change this to a 4 way `if-else if` since it seems this is not much better.  For the latter list in `ExceptionTypes.cpp`, I'll try a map approach (similar to `start_element_map_` and `end_element_map_` in CacheXmlParser